File: packages/intellij-plugin/src/test/kotlin/ai/astha/scantheplanet/idea/contracts/PromptContractsTest.kt
package ai.astha.scantheplanet.idea.contracts

import ai.astha.scantheplanet.idea.scanner.prompt.PromptStrings
import org.junit.Assert.assertEquals
import org.junit.Test
import java.nio.file.Files
import java.nio.file.Path

class PromptContractsTest {
    @Test
    fun systemPromptMatchesContracts() {
        val expected = readContract("prompts/system.prompt.txt")
        assertEquals(expected, PromptStrings.SYSTEM_PROMPT)
    }

    @Test
    fun batchPromptMatchesContracts() {
        val expected = readContract("prompts/batch.prompt.txt")
        assertEquals(expected, PromptStrings.BATCH_SYSTEM_PROMPT)
    }

    @Test
    fun jsonPrefillMatchesContracts() {
        val expected = readContract("prompts/json.prefill.txt")
        assertEquals(expected, PromptStrings.JSON_PREFILL)
    }

    private fun readContract(relativePath: String): String {
        val base = resolveContractsDir()
        val path = base.resolve(relativePath)
        return Files.readString(path).trimEnd()
    }

    private fun resolveContractsDir(): Path {
        val cwd = Path.of("").toAbsolutePath()
        val candidates = listOf(
            cwd.resolve("..").resolve("contracts"),
            cwd.resolve("packages").resolve("contracts"),
            cwd.resolve("..").resolve("..").resolve("packages").resolve("contracts")
        )
        for (candidate in candidates) {
            if (Files.isDirectory(candidate)) {
                return candidate
            }
        }
        throw IllegalStateException("contracts directory not found from $cwd")
    }
}

File: packages/core/tests/promptContracts.test.ts
import { readFileSync } from 'node:fs';
import { resolve } from 'node:path';
import {
  SYSTEM_PROMPT,
  BATCH_SYSTEM_PROMPT,
  JSON_PREFILL,
} from '../src/prompt/PromptBuilder.js';

function readContract(name: string): string {
  const contractsDir = resolve(__dirname, '..', '..', 'contracts', 'prompts');
  return readFileSync(resolve(contractsDir, name), 'utf8').trimEnd();
}

test('system prompt matches contracts', () => {
  expect(SYSTEM_PROMPT).toBe(readContract('system.prompt.txt'));
});

test('batch prompt matches contracts', () => {
  expect(BATCH_SYSTEM_PROMPT).toBe(readContract('batch.prompt.txt'));
});

test('json prefill matches contracts', () => {
  expect(JSON_PREFILL).toBe(readContract('json.prefill.txt'));
});
